### PR TITLE
Add a parameter to choose between resolving links or not

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

**Review by t20100 — APPROVED** [review]
LGTM
Some tests would be nice.

File: h5grove/utils.py [hunk 1]
```diff
@@ -9,15 +9,20 @@ def attr_metadata(attrId: h5py.h5a.AttrID) -> dict:
...
 
     link = h5file.get(path, getlink=True)
     if isinstance(link, h5py.ExternalLink) or isinstance(link, h5py.SoftLink):
-        try:
-            return h5file[path]
-        except (OSError, KeyError):
+        if resolve_links:
+            try:
+                return h5file[path]
+            except (OSError, KeyError):
```
Comment (t20100) [inline h5grove/utils.py:23]: Do you want to get the link info in case of an error rather than an error since the link info can be retrieved otherwise now?

Comment (t20100) [reply h5grove/utils.py:23]: Or would that be useful/possible to get the link info along with the metadata of the `Group`, `Dataset` anyway through e.g. a "link" in ("hard", "soft", "external")?

**Comment by t20100** [comment]
Thanks for the test!

File: h5grove/utils.py (hunk 1 above)
Comment (loichuder) [reply h5grove/utils.py:23]: Good ideas but I would keep this for another PR :+1:

Comment (t20100) [reply h5grove/utils.py:23]: OK ! We indeed do not need to care about backward compatibility for now.

**Review by t20100 — APPROVED** [review]

**Pull request merged**